changes/2024-TBD_cache-across-hierarchy-keyrings/change.md

```diff
@@ -144,7 +145,210 @@ is to make the Hierarchy Keyring
 "Easy to Use" in a multiple KMS Relationship
 environment.
 
-## Relating Cache Entries (Branch Keys) to KMS Relationships
+## Summary
+
+Relevant sim [ticket](https://sim.amazon.com/issues/CrypTool-5311).
+
+To improve the MPL Consumer data key caching experience,
+when using the Hierarchy Keyring,
+we need to allow caching across KeyStores/KMS Clients/KMS Keys.
+
+To facilitate Caching across KeyStores/KMS Clients/KMS Keys,
+we MUST break the Cryptographic Materials Cache (CMC)
+out of the Hierarchy Keyring.
+
+By allowing MPL Consumers to provide an already initialized CMC
+to the Hierarchy Keyring at construction,
+the CMC MAY cache Branch Keys protected by different
+KMS Relationships.
+
+This simplifies Multiple KMS Relationship MPL Consumers,
+as they do not need to stand up LRU Cache of Hierarchy Keyrings.
+
+Instead, they may maintain one CMC.
+They still create a Hierarchy Keyring instance per KMS Relationship,
+and they MUST use the correct Keyring to retrieve material
+from the Cache.
+
+But they only need not maintain the common cache.
+
+Additionally, even though there is no CachingCMM in dafny right
+now, in addition to allowing a shared cache between multiple
+Hierarchy Keyrings, we want to create cache identifiers such
+that multiple CachingCMMs and multiple Hierarchy Keyrings can
+simultaneously use a single shared cache.
+
+## Requirements
+
+- We must allow caching across KeyStores/KMS Clients/KMS Keys
+  for multiple Hierarchy Keyrings and CachingCMMs.
+  For this, we need to strategically update the cache identifiers
+  for all materials ([background](#background)) in the CMC.
+  The [Issues and Alternatives section](#issues-and-alternatives)
+  discusses this.
+
+## Out of Scope
+
+- Supporting a shared CMC for multiple KMS relationships
+  in the **DB-ESDK** to fetch BeaconKeyMaterials.
+
+## Issues and Alternatives:
+
+Before looking at the issues, please take a quick look at
+[Hierarchy Keyring Cache Identifier formula](#hierarchy-keyring-cache-identifier-formula),
+[CachingCMM Cache Identifier formula](#cachingcmm-cache-identifier-formula), and
+[DB-ESDK Cache Identifier formula](#db-esdk-cache-identifier-formula) for more context.
```

Comment:
Copy these over.
Once this work lands, the content will be different.
If you need the current formula's for this change doc to make sense, 
you will need to statically define them.

Reply:
I copied the CachingCMM cache identifier formula. I've tried to reference the current commit wherever possible. Thanks for pointing this out.